### PR TITLE
Move to AWS v3 library

### PR DESCRIPTION
**Pull request by jjlimepoint**
As the world turns, so do library versions.  Move to AWS v3 - note that this has ONLY been tested on my integration test rig, which means only ec2 / custom networking / vpc, s3, and elb have been tested, however I'm pretty confident in the safety given that those required no changes beyond the gemfile.

That said, I did bump the major version since v3 aws is a major change that affects gem dependencies.  It would work as v2/v3 allowed, if we wanted to do that, though I haven't done that in this PR due to test coverage issues (I wanted something testable in my environment...)

It probably depends on the PR for provisioning being removed from DK, since DK uses aws v2 (though their PR says they want to use v3 as well, and prov-aws is holding them back, so *shrug*)

Someone other than me should probably test this before merging :)

### CODE REVIEW

**Comment by jjlimepoint** [comment]
Updated to not hard require v3, but rather allow it as an option, and remove some cruft comments that I added to the patch

**Comment by tas50** [comment]
Looks like you can squash this down to a single commit. There's also some Travis failures going on.

**Comment by jjlimepoint** [comment]
@tas50 both facts - the travis one caught a corner case which I had not.  huzzah for tests.  (also squashed now)

**Comment by tas50** [comment]
@jjlimepoint I'd really like to migrate this gem to the V3 library, but I'd like to make sure we use the individual gems instead of the monolithic gem. There's a lot of code we can shed off if we move to the appropriate v3 gems. Is that something you want to tackle or should I close this out and take a stab myself?

**Comment by jjlimepoint** [comment]
I can do that tomorrow most likely - I mostly avoided it initially in a vain attempt to maintain v2 compatibility - but i'm not super inclined to care about support for v2 if no-one else is!

**Comment by tas50** [comment]
I'm working to get V3 into kitchen-ec2 and we'll need to do the same thing with inspec. It requires all 3 to land at the same time. Total pain, but we have to do it sometime.

File: chef-provisioning-aws.gemspec [hunk 1]
```diff
@@ -17,7 +17,22 @@ Gem::Specification.new do |s|
 
   s.add_dependency "chef-provisioning", ">= 1.0", "< 3.0"
 
-  s.add_dependency "aws-sdk", [">= 2.2.18", "< 3.0"]
+  # all currently supported AWS services
+  s.add_dependency "aws-sdk-core", [">= 3.0", "< 4.0"]
+  s.add_dependency "aws-sdk-ec2", [">= 1.42.0", "< 2.0"]
+  s.add_dependency "aws-sdk-s3", [">= 1.17.0", "< 2.0"]
+  s.add_dependency "aws-sdk-rds", [">= 1.0", "< 4.0"]
+  s.add_dependency "aws-sdk-route53", [">= 1.0", "< 4.0"]
```
Comment (tas50) [inline chef-provisioning-aws.gemspec]: The resource deps should all be ~> 1.0 since Amazon is on 1.x right now.

File: lib/chef/provider/aws_key_pair.rb [hunk 2]
```diff
@@ -1,6 +1,5 @@
 require "chef/provider/lwrp_base"
 require "chef/provisioning/aws_driver/aws_provider"
-require "aws-sdk"
```
Comment (tas50) [inline lib/chef/provider/aws_key_pair.rb]: This should be require "aws-sdk-ec2"

File: chef-provisioning-aws.gemspec (hunk 1 above)
Comment (jjlimepoint) [reply chef-provisioning-aws.gemspec]: Good catch - thanks for that!

File: lib/chef/provider/aws_key_pair.rb (hunk 2 above)
Comment (jjlimepoint) [reply lib/chef/provider/aws_key_pair.rb]: At the time I actually removed this deliberatly, since the class doesn't _actually_ use anything from aws-sdk-* directly at all, but rather the things provided by aws_provider.  Still, not real harm in specifying it.

**Comment by cgunasree08** [comment]
This issue or pull request is being automatically closed because the repository is being archived as part of the [Repository Standardization Initiative](https://github.com/chef-boneyard/oss-repo-standardization-2025).
